From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <46881115+Elephant1214@users.noreply.github.com>
Date: Mon, 14 Apr 2025 15:43:03 -0600
Subject: [PATCH] Rebrand


diff --git a/net/minecraft/CrashReport.java b/net/minecraft/CrashReport.java
index 18e63dca2a315ad01fa5b80ec126f4eac71f45fd..6334f93b8355219b3eb69c36f870f0ffc683f36d 100644
--- a/net/minecraft/CrashReport.java
+++ b/net/minecraft/CrashReport.java
@@ -30,7 +30,7 @@ public class CrashReport {
     private boolean trackingStackTrace = true;
     private StackTraceElement[] uncategorizedStackTrace = new StackTraceElement[0];
     private final SystemReport systemReport = new SystemReport();
-    private List<String> extraInfo = List.of("", "DO NOT REPORT THIS TO PAPER! REPORT TO LEAF INSTEAD!", ""); // Leaf - Rebrand
+    private List<String> extraInfo = List.of("", "DO NOT REPORT THIS TO PAPER OR LEAF! REPORT TO ZIRKONIUM INSTEAD!", ""); // Leaf - Rebrand // Zirkonium - Rebrand
 
     public CrashReport(String title, Throwable exception) {
         io.papermc.paper.util.StacktraceDeobfuscator.INSTANCE.deobfuscateThrowable(exception); // Paper
diff --git a/net/minecraft/world/damagesource/DamageSource.java b/net/minecraft/world/damagesource/DamageSource.java
index 5a663f66820d95449ccf9117e72159a10bcd04d6..e0d8d53d5133a3ef9be8f276333a70d3a6d67ee6 100644
--- a/net/minecraft/world/damagesource/DamageSource.java
+++ b/net/minecraft/world/damagesource/DamageSource.java
@@ -75,7 +75,7 @@ public class DamageSource {
 
     public DamageSource eventEntityDamager(final Entity entity) {
         if (this.directEntity != null) {
-            throw new IllegalStateException("Cannot set an event damager when a direct entity is already set (if you think this is a Leaf bug, please report it at https://github.com/Winds-Studio/Leaf/issues)"); // Leaf - Rebrand
+            throw new IllegalStateException("Cannot set an event damager when a direct entity is already set (if you think this is a Zirkonium bug, please report it at https://github.com/ZirkonNetwork/Zirkonium/issues)"); // Leaf - Rebrand // Zirkonium - Rebrand
         }
         final DamageSource damageSource = this.copy();
         damageSource.eventEntityDamager = entity;
diff --git a/net/minecraft/world/level/chunk/storage/RegionFileStorage.java b/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
index 3634d68c77ee99567b51a9a848d7cbbb218a0ba7..8f23251dc70eb8612170f40363c10b2f010efe1a 100644
--- a/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
+++ b/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
@@ -308,7 +308,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
 
     // Paper start
     private static void printOversizedLog(String msg, Path file, int x, int z) {
-        org.apache.logging.log4j.LogManager.getLogger().fatal(msg + " (" + file.toString().replaceAll(".+[\\\\/]", "") + " - " + x + "," + z + ") Go clean it up to remove this message. /minecraft:tp " + (x<<4)+" 128 "+(z<<4) + " - DO NOT REPORT THIS TO PAPER OR LEAF - You may ask for help on Discord, but do not file an issue. These error messages can not be removed. - If you think this is a Leaf bug, please report it at https://github.com/Winds-Studio/Leaf/issues)"); // Leaf - Rebrand
+        org.apache.logging.log4j.LogManager.getLogger().fatal(msg + " (" + file.toString().replaceAll(".+[\\\\/]", "") + " - " + x + "," + z + ") Go clean it up to remove this message. /minecraft:tp " + (x<<4)+" 128 "+(z<<4) + " - DO NOT REPORT THIS TO PAPER, LEAF, OR ZIRKONIUM - You may ask for help on Discord, but do not file an issue. These error messages can not be removed. - If you think this is a Zirkonium bug, please report it at https://github.com/ZirkonNetwork/Zirkonium/issues)"); // Leaf - Rebrand // Zirkonium - Rebrand
     }
 
     private static CompoundTag readOversizedChunk(RegionFile regionfile, ChunkPos chunkCoordinate) throws IOException {
