From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <46881115+Elephant1214@users.noreply.github.com>
Date: Tue, 25 Feb 2025 20:30:23 -0700
Subject: [PATCH] Rebrand


diff --git a/net/minecraft/CrashReport.java b/net/minecraft/CrashReport.java
index 5a9173b6039ee8404529a04904c5f9e9343d74d8..e7920897105021da5170aef434a37ee9080bf072 100644
--- a/net/minecraft/CrashReport.java
+++ b/net/minecraft/CrashReport.java
@@ -30,7 +30,7 @@ public class CrashReport {
     private boolean trackingStackTrace = true;
     private StackTraceElement[] uncategorizedStackTrace = new StackTraceElement[0];
     private final SystemReport systemReport = new SystemReport();
-    private List<String> extraInfo = List.of("", "DO NOT REPORT THIS TO PAPER OR GALE! REPORT TO LEAF INSTEAD!", ""); // Leaf - Rebrand
+    private List<String> extraInfo = List.of("", "DO NOT REPORT THIS TO PAPER, GALE, OR LEAF! REPORT TO ZIRKONIUM INSTEAD!", ""); // Leaf - Rebrand // Zirkonium - Rebrand
 
     public CrashReport(String title, Throwable exception) {
         io.papermc.paper.util.StacktraceDeobfuscator.INSTANCE.deobfuscateThrowable(exception); // Paper
diff --git a/net/minecraft/world/damagesource/DamageSource.java b/net/minecraft/world/damagesource/DamageSource.java
index bf9520bf4f314dacc66db9a7d30dbcc239db2c7d..5e7dc3db45372eb3a83aa8073f4d16be7fc64573 100644
--- a/net/minecraft/world/damagesource/DamageSource.java
+++ b/net/minecraft/world/damagesource/DamageSource.java
@@ -75,7 +75,7 @@ public class DamageSource {
 
     public DamageSource eventEntityDamager(final Entity entity) {
         if (this.directEntity != null) {
-            throw new IllegalStateException("Cannot set an event damager when a direct entity is already set (report a bug to Paper, if you think this is a Leaf bug, please report it at https://github.com/Winds-Studio/Leaf/issues)"); // Gale - branding changes // Leaf - Rebrand
+            throw new IllegalStateException("Cannot set an event damager when a direct entity is already set (report a bug to Paper, if you think this is a Zirkonium bug, please report it at https://github.com/ZirkonNetwork/Zirkonium/issues)"); // Gale - branding changes // Leaf - Rebrand // Zirkonium - Rebrand
         }
         final DamageSource damageSource = this.copy();
         damageSource.eventEntityDamager = entity;
diff --git a/net/minecraft/world/level/chunk/storage/RegionFileStorage.java b/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
index 5d278003313fae60a5948eab207b5257cd8bb4bf..26a1a2909975310b4b99b7d01ecc59a59ae2a7dc 100644
--- a/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
+++ b/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
@@ -294,7 +294,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
         We do not want people to report thread issues to Paper,
         but we do want people to report thread issues to Gale.
         */
-        org.apache.logging.log4j.LogManager.getLogger().fatal(msg + " (" + file.toString().replaceAll(".+[\\\\/]", "") + " - " + x + "," + z + ") Go clean it up to remove this message. /minecraft:tp " + (x<<4)+" 128 "+(z<<4) + " - DO NOT REPORT THIS TO PAPER OR LEAF - You may ask for help on Discord, but do not file an issue. These error messages can not be removed. - If you think this is a Leaf bug, please report it at https://github.com/Winds-Studio/Leaf/issues)"); // Leaf - Rebrand
+        org.apache.logging.log4j.LogManager.getLogger().fatal(msg + " (" + file.toString().replaceAll(".+[\\\\/]", "") + " - " + x + "," + z + ") Go clean it up to remove this message. /minecraft:tp " + (x<<4)+" 128 "+(z<<4) + " - DO NOT REPORT THIS TO PAPER OR ZIRKONIUM - You may ask for help on Discord, but do not file an issue. These error messages can not be removed. - If you think this is a Zirkonium bug, please report it at https://github.com/ZirkonNetwork/Zirkonium/issues)"); // Leaf - Rebrand // Zirkonium - Rebrand
         // Gale end - branding changes
     }
 
