From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <46881115+Elephant1214@users.noreply.github.com>
Date: Mon, 14 Apr 2025 15:57:22 -0600
Subject: [PATCH] Rebrand


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index c198304386e83edc2ec582fdf66d97d624004771..5208b70b66cc6a75bac5b1eefa6888032d3e129d 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -41,6 +41,13 @@ public interface ServerBuildInfo {
     Key BRAND_LEAF_ID = Key.key("winds-studio", "leaf");
     // Leaf end - Rebrand
 
+    // Zirkonium start - Rebrand
+    /**
+     * The brand id for Zirkonium.
+     */
+    Key BRAND_ZIRKONIUM_ID = Key.key("zirkon-network", "zirkonium");
+    // Zirkonium end - Rebrand
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
