From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <46881115+Elephant1214@users.noreply.github.com>
Date: Fri, 17 May 2024 19:15:41 -0600
Subject: [PATCH] Rebrand


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 7196594e07af19a14c320d77df893978525fe386..72112c2b12bca13bd2e8a2052b5f693fb9ae4e30 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -32,6 +32,13 @@ public interface ServerBuildInfo {
      */
     Key BRAND_PURPUR_ID = Key.key("purpurmc", "purpur");
     // Purpur end
+    
+    // Zirkonium start - Rebrand
+    /**
+     * The brand id for Zirkonium.
+     */
+    Key BRAND_ZIRKONIUM_ID = Key.key("zirkonnetwork", "zirkonium");
+    // Zirkonium end - Rebrand
 
     /**
      * Gets the {@code ServerBuildInfo}.
