From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <46881115+Elephant1214@users.noreply.github.com>
Date: Fri, 17 May 2024 19:15:41 -0600
Subject: [PATCH] Rebrand


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 7107eabf02d9d9b69d97c978ba421ad52031ec3f..2abc5dede1cf3e5407702d5dfe4527abbffbd577 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -25,6 +25,13 @@ public interface ServerBuildInfo {
     Key BRAND_PURPUR_ID = Key.key("purpurmc", "purpur");
     // Purpur end
 
+    // Zirkonium start
+    /**
+     * The brand id for Zirkonium.
+     */
+    Key BRAND_ZIRKONIUM_ID = Key.key("zirkonnetwork", "zirkonium");
+    // Zirkonium end
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
