From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <46881115+Elephant1214@users.noreply.github.com>
Date: Sat, 24 Dec 2022 15:58:13 -0700
Subject: [PATCH] Fix the server buildscript


diff --git a/build.gradle.kts b/build.gradle.kts
index 89f21e21efc5f5ab99d1586f4f3b8f2f3c48ac06..930d69d6f495d41387dc346bce726ddc07d785ba 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -12,7 +12,7 @@ configurations.named(log4jPlugins.compileClasspathConfigurationName) {
 val alsoShade: Configuration by configurations.creating
 
 dependencies {
-    implementation(project(":purpur-api")) // Purpur
+    implementation(project(":zirkonium-api")) // Purpur // Zirkonium
     // Paper start
     implementation("org.jline:jline-terminal-jansi:3.21.0")
     implementation("net.minecrell:terminalconsoleappender:1.3.0")
