From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <nathangetsmail@gmail.com>
Date: Sat, 24 Dec 2022 15:58:13 -0700
Subject: [PATCH] Fix the server buildscript


diff --git a/build.gradle.kts b/build.gradle.kts
index 790835a0c4889d17b4efd06cb463744973642a48..030581d4ee6adc4593e5ca95f8825766cd3c32a9 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -14,7 +14,7 @@ val alsoShade: Configuration by configurations.creating
 
 dependencies {
     // Purpur start
-    implementation(project(":purpur-api"))
+    implementation(project(":Zirkonium-API"))
     implementation("io.papermc.paper:paper-mojangapi:${project.version}") {
         exclude("io.papermc.paper", "paper-api")
     }
