From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <46881115+Elephant1214@users.noreply.github.com>
Date: Sat, 24 Dec 2022 15:58:13 -0700
Subject: [PATCH] Fix the server buildscript


diff --git a/build.gradle.kts b/build.gradle.kts
index 77602e4445b1b45a17e5552ac6a769e0ed46ebc7..f785d43af00af75011b66ef152c6e20e8daa43a7 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -25,7 +25,7 @@ abstract class MockitoAgentProvider : CommandLineArgumentProvider {
 // Paper end - configure mockito agent that is needed in newer java versions
 
 dependencies {
-    implementation(project(":purpur-api")) // Pufferfish // Paper // Purpur
+    implementation(project(":zirkonium-api")) // Pufferfish // Paper // Purpur // Zirkonium
     // Paper start
     implementation("org.jline:jline-terminal-ffm:3.27.1") // use ffm on java 22+
     implementation("org.jline:jline-terminal-jni:3.27.1") // fall back to jni on java 21
