From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <nathangetsmail@gmail.com>
Date: Sat, 24 Dec 2022 15:58:13 -0700
Subject: [PATCH] Fix the server buildscript


diff --git a/build.gradle.kts b/build.gradle.kts
index 9c01005751c0088f560f96401cdfdebbbda4e7ec..bc11029baa961543d8b7eec0c432972d1dad95f4 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -7,8 +7,13 @@ plugins {
 }
 
 dependencies {
-    implementation(project(":purpur-api")) // Purpur
+    implementation(project(":Zirkonium-API")) // Zirkonium
     implementation("io.papermc.paper:paper-mojangapi:1.19.4-R0.1-SNAPSHOT") // Purpur
+    // Pufferfish start
+    implementation("io.papermc.paper:paper-mojangapi:1.19.3-R0.1-SNAPSHOT") {
+        exclude("io.papermc.paper", "paper-api")
+    }
+    // Pufferfish end
     // Paper start
     implementation("org.jline:jline-terminal-jansi:3.21.0")
     implementation("net.minecrell:terminalconsoleappender:1.3.0")
