From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <nathangetsmail@gmail.com>
Date: Sat, 24 Dec 2022 15:58:13 -0700
Subject: [PATCH] Fix the server buildscript


diff --git a/build.gradle.kts b/build.gradle.kts
index 0c92d10f334f82d196e065c0811d0c1f48887377..fa323c29499ba6d4149c41feba03992b8ef9fea6 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -14,7 +14,7 @@ val alsoShade: Configuration by configurations.creating
 
 dependencies {
     // Purpur start
-    implementation(project(":purpur-api"))
+    implementation(project(":Zirkonium-API"))
     implementation("io.papermc.paper:paper-mojangapi:${project.version}") {
         exclude("io.papermc.paper", "paper-api")
     }
