From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <nathangetsmail@gmail.com>
Date: Sat, 24 Dec 2022 17:00:43 -0700
Subject: [PATCH] Implement No Chat Reports


diff --git a/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java b/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
index ba585089164b7e6bc48fd28e8d389bbc95c83673..f0c485c2b6ede0e617ae0e75c780992d14018516 100644
--- a/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
+++ b/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
@@ -40,6 +40,8 @@ public class ZirkoniumConfig {
     }
 
     static void readConfig() {
+        NoChatReports.noChatReportsSettings();
+
         try {
             config.save(CONFIG_FILE);
         } catch (IOException ex) {
@@ -66,7 +68,15 @@ public class ZirkoniumConfig {
         return config.getInt(path, config.getInt(path));
     }
 
-    public static class Gameplay {
-        public static int maxAnvilLevelCost = 40;
+    public static class NoChatReports {
+        public static boolean noChatReports = true;
+        public static boolean addQueryData = true;
+        public static boolean convertToSystemMessage = true;
+
+        public static void noChatReportsSettings() {
+            noChatReports = getBoolean("no-chat-reports.enabled", noChatReports);
+            addQueryData = getBoolean("no-chat-reports.add-query-data", addQueryData) && noChatReports;
+            convertToSystemMessage = getBoolean("no-chat-reports.convert-to-system-message", convertToSystemMessage) && noChatReports;
+        }
     }
 }
diff --git a/src/main/java/net/minecraft/network/FriendlyByteBuf.java b/src/main/java/net/minecraft/network/FriendlyByteBuf.java
index 1f4b64a5f812376c499c98cb4be62469bd0b7dbe..b3c780673783542aa3b9d7e2bbf3f412bf431c59 100644
--- a/src/main/java/net/minecraft/network/FriendlyByteBuf.java
+++ b/src/main/java/net/minecraft/network/FriendlyByteBuf.java
@@ -145,17 +145,40 @@ public class FriendlyByteBuf extends ByteBuf {
     }
 
     public <T> T readJsonWithCodec(Codec<T> codec) {
+        // Zirkonium start - No Chat Reports
+        if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.NoChatReports.noChatReports &&
+                codec == net.minecraft.network.protocol.status.ServerStatus.CODEC) {
+            JsonElement jsonElement = GsonHelper.fromJson(FriendlyByteBuf.GSON, this.readUtf(), JsonElement.class);
+            DataResult<T> dataResult = codec.parse(JsonOps.INSTANCE, jsonElement);
+            T result = Util.getOrThrow(dataResult, string -> new DecoderException("Failed to decode json: " + string));
+
+            if (jsonElement.getAsJsonObject().has("preventsChatReports")) {
+                ((net.minecraft.network.protocol.status.ServerStatus) result).setPreventsChatReports(jsonElement.getAsJsonObject()
+                        .get("preventsChatReports")
+                        .getAsBoolean());
+            }
+            return result;
+        } // Zirkonium end - No Chat Reports
+
         JsonElement jsonelement = (JsonElement) GsonHelper.fromJson(FriendlyByteBuf.GSON, this.readUtf(), JsonElement.class);
         DataResult<T> dataresult = codec.parse(JsonOps.INSTANCE, jsonelement);
-
         return Util.getOrThrow(dataresult, (s) -> {
             return new DecoderException("Failed to decode json: " + s);
         });
     }
 
     public <T> void writeJsonWithCodec(Codec<T> codec, T value) {
-        DataResult<JsonElement> dataresult = codec.encodeStart(JsonOps.INSTANCE, value);
+        // Zirkonium start - No Chat Reports
+        if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.NoChatReports.addQueryData &&
+                codec == net.minecraft.network.protocol.status.ServerStatus.CODEC) {
+            DataResult<JsonElement> dataresult = codec.encodeStart(JsonOps.INSTANCE, value);
+            JsonElement element = Util.getOrThrow(dataresult, string -> new EncoderException("Failed to encode: " + string + " " + value));
+            element.getAsJsonObject().addProperty("preventsChatReports", true);
+            this.writeUtf(GSON.toJson(element));
+            return;
+        } // Zirkonium end - No Chat Reports
 
+        DataResult<JsonElement> dataresult = codec.encodeStart(JsonOps.INSTANCE, value);
         this.writeUtf(FriendlyByteBuf.GSON.toJson((JsonElement) Util.getOrThrow(dataresult, (s) -> {
             return new EncoderException("Failed to encode: " + s + " " + value);
         })));
diff --git a/src/main/java/net/minecraft/network/protocol/game/ClientboundSystemChatPacket.java b/src/main/java/net/minecraft/network/protocol/game/ClientboundSystemChatPacket.java
index 1387e3597c43fd652f2fc82ca6fc2e83039604e2..8a126cde22eeb76d638fc5583ed72d38a91834f8 100644
--- a/src/main/java/net/minecraft/network/protocol/game/ClientboundSystemChatPacket.java
+++ b/src/main/java/net/minecraft/network/protocol/game/ClientboundSystemChatPacket.java
@@ -8,7 +8,7 @@ import net.minecraft.network.protocol.Packet;
 // Spigot start
 public record ClientboundSystemChatPacket(@javax.annotation.Nullable net.kyori.adventure.text.Component adventure$content, @javax.annotation.Nullable String content, boolean overlay) implements Packet<ClientGamePacketListener> { // Paper - Adventure
 
-    @io.papermc.paper.annotation.DoNotUse // Paper - No locale context
+    @io.papermc.paper.annotation.DoNotUse
     public ClientboundSystemChatPacket(Component content, boolean overlay) {
         this(null, Component.Serializer.toJson(content), overlay); // Paper - Adventure
     }
diff --git a/src/main/java/net/minecraft/network/protocol/game/ServerboundChatCommandPacket.java b/src/main/java/net/minecraft/network/protocol/game/ServerboundChatCommandPacket.java
index b8c1f3b9afddc87d56541c8af63cffecfcdd2653..19215e203a56046be7718802097569d6addc65d4 100644
--- a/src/main/java/net/minecraft/network/protocol/game/ServerboundChatCommandPacket.java
+++ b/src/main/java/net/minecraft/network/protocol/game/ServerboundChatCommandPacket.java
@@ -24,4 +24,10 @@ public record ServerboundChatCommandPacket(String command, Instant timeStamp, lo
     public void handle(ServerGamePacketListener listener) {
         listener.handleChatCommand(this);
     }
+
+    @Override // Zirkonium start - No Chat Reports
+    public ArgumentSignatures argumentSignatures() {
+        if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.NoChatReports.noChatReports) return ArgumentSignatures.EMPTY;
+        return this.argumentSignatures;
+    } // Zirkonium end - No Chat Reports
 }
diff --git a/src/main/java/net/minecraft/network/protocol/game/ServerboundChatPacket.java b/src/main/java/net/minecraft/network/protocol/game/ServerboundChatPacket.java
index d1d2fc0c57523c1abf1e8bfec913c78927c3dafc..6be61a19395e897537d774365587238c960a58a2 100644
--- a/src/main/java/net/minecraft/network/protocol/game/ServerboundChatPacket.java
+++ b/src/main/java/net/minecraft/network/protocol/game/ServerboundChatPacket.java
@@ -25,4 +25,10 @@ public record ServerboundChatPacket(String message, Instant timeStamp, long salt
     public void handle(ServerGamePacketListener listener) {
         listener.handleChat(this);
     }
+
+    @Override // Zirkonium start - No Chat Reports
+    public @Nullable MessageSignature signature() {
+        if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.NoChatReports.noChatReports) return null;
+        return this.signature;
+    } // Zirkonium end - No Chat Reports
 }
diff --git a/src/main/java/net/minecraft/network/protocol/game/ServerboundChatSessionUpdatePacket.java b/src/main/java/net/minecraft/network/protocol/game/ServerboundChatSessionUpdatePacket.java
index 9d6db4a378036559efab91c8b7dcf2a6b0c2cce6..0e2fa14eda69ba5aa3ee4d6901f43b8bbcec57e8 100644
--- a/src/main/java/net/minecraft/network/protocol/game/ServerboundChatSessionUpdatePacket.java
+++ b/src/main/java/net/minecraft/network/protocol/game/ServerboundChatSessionUpdatePacket.java
@@ -16,6 +16,7 @@ public record ServerboundChatSessionUpdatePacket(RemoteChatSession.Data chatSess
 
     @Override
     public void handle(ServerGamePacketListener listener) {
+        if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.NoChatReports.noChatReports) return; // Zirkonium - No Chat Reports
         listener.handleChatSessionUpdate(this);
     }
 }
diff --git a/src/main/java/net/minecraft/network/protocol/status/ServerStatus.java b/src/main/java/net/minecraft/network/protocol/status/ServerStatus.java
index 98ea56cbc8de5a5694b4f4d9a2f25811fdecba04..8627a769d8858cc501f70a5a33a0efcb9caafa6a 100644
--- a/src/main/java/net/minecraft/network/protocol/status/ServerStatus.java
+++ b/src/main/java/net/minecraft/network/protocol/status/ServerStatus.java
@@ -62,4 +62,27 @@ public record ServerStatus(Component description, Optional<ServerStatus.Players>
             return new ServerStatus.Version(worldVersion.getName(), worldVersion.getProtocolVersion());
         }
     }
-}
+
+    // Zirkonium start - No Chat Reports
+    private static boolean preventsChatReports;
+
+    @Override
+    public boolean enforcesSecureChat() {
+        if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.NoChatReports.convertToSystemMessage) {
+            return true;
+        }
+        return this.enforcesSecureChat;
+    }
+
+    public boolean preventsChatReports() {
+        if (this.version.isPresent() && this.version.get().protocol < 759 && this.version.get().protocol > 0) {
+            return true;
+        }
+        return preventsChatReports;
+    }
+
+    public void setPreventsChatReports(boolean prevents) {
+        preventsChatReports = prevents;
+    }
+    // Zirkonium end
+}
\ No newline at end of file
diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index a46491caf247c92ca79563384a99060bc90c97bc..68b38a0e2e821ea79cad15c5d688f5e649b441f8 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -674,7 +674,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
     @Override
     public boolean enforceSecureProfile() {
         DedicatedServerProperties dedicatedserverproperties = this.getProperties();
-
+        if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.NoChatReports.noChatReports) return false; // Zirkonium - No Chat Reports
         return dedicatedserverproperties.enforceSecureProfile && dedicatedserverproperties.onlineMode && this.services.profileKeySignatureValidator() != null;
     }
 
diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 255b422078880b46dfa6f5140f6c92fef057d09e..18cf72bd1bc0352320d8437f53a41aeb67177ed5 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -2293,10 +2293,24 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
 
     @Override
     public void send(Packet<?> packet) {
+        // Zirkonium start - No Chat Reports
+        if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.NoChatReports.convertToSystemMessage && packet instanceof ClientboundPlayerChatPacket chat) {
+            packet = new ClientboundSystemChatPacket(null, Component.Serializer.toJson(chat.chatType().resolve(this.player.level().registryAccess())
+                    .get().decorate(chat.unsignedContent() != null ? chat.unsignedContent()
+                            : Component.literal(chat.body().content()))), false);
+            this.send(packet);
+            return;
+        }
         this.send(packet, (PacketSendListener) null);
     }
 
     public void send(Packet<?> packet, @Nullable PacketSendListener callbacks) {
+        if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.NoChatReports.convertToSystemMessage && packet instanceof ClientboundPlayerChatPacket playerChat && callbacks != null) {
+            this.send(playerChat);
+            return;
+        }
+        // Zirkonium end - No Chat Reports
+
         // CraftBukkit start
         if (packet == null || this.processedDisconnect) { // Spigot
             return;
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 824f31b68b38f2f8642fb9d59a123cfdaffbb7b2..1fa2e9bbc51e542380577d5328af711fc2ffbd85 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -1,6 +1,5 @@
 package net.minecraft.server.players;
 
-import co.aikar.timings.MinecraftTimings;
 import com.google.common.collect.Lists;
 import com.google.common.collect.Maps;
 import com.google.common.collect.Sets;
@@ -118,7 +117,6 @@ import org.bukkit.Location;
 import org.bukkit.craftbukkit.CraftServer;
 import org.bukkit.craftbukkit.CraftWorld;
 import org.bukkit.craftbukkit.entity.CraftPlayer;
-import org.bukkit.craftbukkit.util.CraftChatMessage;
 import org.bukkit.craftbukkit.util.CraftLocation;
 import org.bukkit.entity.Player;
 import org.bukkit.event.player.PlayerChangedWorldEvent;
@@ -127,7 +125,6 @@ import org.bukkit.event.player.PlayerLoginEvent;
 import org.bukkit.event.player.PlayerQuitEvent;
 import org.bukkit.event.player.PlayerRespawnEvent;
 import org.bukkit.event.player.PlayerRespawnEvent.RespawnReason;
-import org.bukkit.event.player.PlayerSpawnChangeEvent;
 // CraftBukkit end
 
 public abstract class PlayerList {
@@ -1479,6 +1476,7 @@ public abstract class PlayerList {
     }
 
     public boolean verifyChatTrusted(PlayerChatMessage message) { // Paper - private -> public
+        if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.NoChatReports.noChatReports) return true; // Zirkonium - No Chat Reports
         return message.hasSignature() && !message.hasExpiredServer(Instant.now());
     }
 
