From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <nathangetsmail@gmail.com>
Date: Sat, 24 Dec 2022 17:00:43 -0700
Subject: [PATCH] Implement No Chat Reports


diff --git a/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java b/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
index 7d35b880d5b58aebd220d83def3b997e15029b09..76ca77707a811c79a6b3a13d31ba1e9f49d20c08 100644
--- a/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
+++ b/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
@@ -41,6 +41,8 @@ public class ZirkoniumConfig {
     }
 
     static void readConfig() {
+        noChatReportsSettings();
+
         try {
             config.save(CONFIG_FILE);
         } catch (IOException ex) {
@@ -62,4 +64,16 @@ public class ZirkoniumConfig {
         config.addDefault(path, def);
         return config.getInt(path, config.getInt(path));
     }
+
+    // Actual config options are below this point
+
+    public static boolean noChatReports = true;
+    public static boolean addQueryData = true;
+    public static boolean convertToSystemMessage = true;
+
+    private static void noChatReportsSettings() {
+        noChatReports = getBoolean("no-chat-reports.enabled", noChatReports);
+        addQueryData = getBoolean("no-chat-reports.add-query-data", addQueryData) && noChatReports;
+        convertToSystemMessage = getBoolean("no-chat-reports.convert-to-system-message", convertToSystemMessage) && noChatReports;
+    }
 }
diff --git a/src/main/java/net/minecraft/network/FriendlyByteBuf.java b/src/main/java/net/minecraft/network/FriendlyByteBuf.java
index bdcfd80f937c34956911373905d66424bbff8e1d..a273042199b7653d55d05b1755539b799ef30263 100644
--- a/src/main/java/net/minecraft/network/FriendlyByteBuf.java
+++ b/src/main/java/net/minecraft/network/FriendlyByteBuf.java
@@ -129,9 +129,23 @@ public class FriendlyByteBuf extends ByteBuf {
     }
 
     public <T> T readJsonWithCodec(Codec<T> codec) {
+        // Zirkonium start - No Chat Reports
+        if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.noChatReports &&
+                codec == net.minecraft.network.protocol.status.ServerStatus.CODEC) {
+            JsonElement jsonElement = GsonHelper.fromJson(FriendlyByteBuf.GSON, this.readUtf(), JsonElement.class);
+            DataResult<T> dataResult = codec.parse(JsonOps.INSTANCE, jsonElement);
+            T result = Util.getOrThrow(dataResult, string -> new DecoderException("Failed to decode json: " + string));
+
+            if (jsonElement.getAsJsonObject().has("preventsChatReports")) {
+                ((net.minecraft.network.protocol.status.ServerStatus) result).setPreventsChatReports(jsonElement.getAsJsonObject()
+                        .get("preventsChatReports")
+                        .getAsBoolean());
+            }
+            return result;
+        } // Zirkonium end - No Chat Reports
+
         JsonElement jsonelement = (JsonElement) GsonHelper.fromJson(FriendlyByteBuf.GSON, this.readUtf(), JsonElement.class);
         DataResult<T> dataresult = codec.parse(JsonOps.INSTANCE, jsonelement);
-
         return Util.getOrThrow(dataresult, (s) -> {
             return new DecoderException("Failed to decode json: " + s);
         });
@@ -144,7 +158,14 @@ public class FriendlyByteBuf extends ByteBuf {
     public <T> void writeJsonWithCodec(Codec<T> codec, T value, int maxLength) {
         // Paper end - Adventure; add max length parameter
         DataResult<JsonElement> dataresult = codec.encodeStart(JsonOps.INSTANCE, value);
-
+        // Zirkonium start - No Chat Reports
+        if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.addQueryData &&
+                codec == net.minecraft.network.protocol.status.ServerStatus.CODEC) {
+            JsonElement element = Util.getOrThrow(dataresult, string -> new EncoderException("Failed to encode: " + string + " " + value));
+            element.getAsJsonObject().addProperty("preventsChatReports", true);
+            this.writeUtf(GSON.toJson(element));
+            return;
+        } // Zirkonium end - No Chat Reports
         this.writeUtf(FriendlyByteBuf.GSON.toJson((JsonElement) Util.getOrThrow(dataresult, (s) -> {
             return new EncoderException("Failed to encode: " + s + " " + value);
         })), maxLength); // Paper - Adventure; add max length parameter
diff --git a/src/main/java/net/minecraft/network/protocol/game/ServerboundChatCommandPacket.java b/src/main/java/net/minecraft/network/protocol/game/ServerboundChatCommandPacket.java
index b8c1f3b9afddc87d56541c8af63cffecfcdd2653..61ab75bc9d7f1914f2ceed472d2d4b460c4cf48e 100644
--- a/src/main/java/net/minecraft/network/protocol/game/ServerboundChatCommandPacket.java
+++ b/src/main/java/net/minecraft/network/protocol/game/ServerboundChatCommandPacket.java
@@ -24,4 +24,9 @@ public record ServerboundChatCommandPacket(String command, Instant timeStamp, lo
     public void handle(ServerGamePacketListener listener) {
         listener.handleChatCommand(this);
     }
+
+    @Override // Zirkonium start - No Chat Reports
+    public ArgumentSignatures argumentSignatures() {
+        return me.elephant1214.zirkonium.configuration.ZirkoniumConfig.noChatReports ? ArgumentSignatures.EMPTY : this.argumentSignatures;
+    } // Zirkonium end - No Chat Reports
 }
diff --git a/src/main/java/net/minecraft/network/protocol/game/ServerboundChatPacket.java b/src/main/java/net/minecraft/network/protocol/game/ServerboundChatPacket.java
index d1d2fc0c57523c1abf1e8bfec913c78927c3dafc..999771c0e4d8751aef25d247d454253549518e15 100644
--- a/src/main/java/net/minecraft/network/protocol/game/ServerboundChatPacket.java
+++ b/src/main/java/net/minecraft/network/protocol/game/ServerboundChatPacket.java
@@ -25,4 +25,9 @@ public record ServerboundChatPacket(String message, Instant timeStamp, long salt
     public void handle(ServerGamePacketListener listener) {
         listener.handleChat(this);
     }
+
+    @Override // Zirkonium start - No Chat Reports
+    public @Nullable MessageSignature signature() {
+        return me.elephant1214.zirkonium.configuration.ZirkoniumConfig.noChatReports ? null : this.signature;
+    } // Zirkonium end - No Chat Reports
 }
diff --git a/src/main/java/net/minecraft/network/protocol/game/ServerboundChatSessionUpdatePacket.java b/src/main/java/net/minecraft/network/protocol/game/ServerboundChatSessionUpdatePacket.java
index 9d6db4a378036559efab91c8b7dcf2a6b0c2cce6..38d7512c7cae0857867eb0a155d3966adbf5b08a 100644
--- a/src/main/java/net/minecraft/network/protocol/game/ServerboundChatSessionUpdatePacket.java
+++ b/src/main/java/net/minecraft/network/protocol/game/ServerboundChatSessionUpdatePacket.java
@@ -16,6 +16,7 @@ public record ServerboundChatSessionUpdatePacket(RemoteChatSession.Data chatSess
 
     @Override
     public void handle(ServerGamePacketListener listener) {
+        if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.noChatReports) return; // Zirkonium - No Chat Reports
         listener.handleChatSessionUpdate(this);
     }
 }
diff --git a/src/main/java/net/minecraft/network/protocol/status/ServerStatus.java b/src/main/java/net/minecraft/network/protocol/status/ServerStatus.java
index c40979e2fa37acb2f3df493395f3ab6e185345d7..dc718ab6ba8d5eecec6eb1db5ba4f24b9590da30 100644
--- a/src/main/java/net/minecraft/network/protocol/status/ServerStatus.java
+++ b/src/main/java/net/minecraft/network/protocol/status/ServerStatus.java
@@ -62,4 +62,27 @@ public record ServerStatus(Component description, Optional<ServerStatus.Players>
             return new ServerStatus.Version(worldVersion.getName(), worldVersion.getProtocolVersion());
         }
     }
-}
+
+    // Zirkonium start - No Chat Reports
+    private static boolean preventsChatReports;
+
+    @Override
+    public boolean enforcesSecureChat() {
+        if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.convertToSystemMessage) {
+            return true;
+        }
+        return this.enforcesSecureChat;
+    }
+
+    public boolean preventsChatReports() {
+        if (this.version.isPresent() && this.version.get().protocol < 759 && this.version.get().protocol > 0) {
+            return true;
+        }
+        return preventsChatReports;
+    }
+
+    public void setPreventsChatReports(boolean prevents) {
+        preventsChatReports = prevents;
+    }
+    // Zirkonium end
+}
\ No newline at end of file
diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index cbc86432a7a96360d7f857b5bcc5a64a35442ce5..d5491686130202b58b36c0ef10829348b94bac12 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -674,9 +674,9 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
     @Override
     public boolean enforceSecureProfile() {
         DedicatedServerProperties dedicatedserverproperties = this.getProperties();
-
-        // Paper start - Add setting for proxy online mode status
-        return dedicatedserverproperties.enforceSecureProfile
+        // Paper start - fix secure profile with proxy online mode
+        return !me.elephant1214.zirkonium.configuration.ZirkoniumConfig.noChatReports // Zirkonium - No Chat Reports
+            && dedicatedserverproperties.enforceSecureProfile
             && io.papermc.paper.configuration.GlobalConfiguration.get().proxies.isProxyOnlineMode()
             && this.services.canValidateProfileKeys();
         // Paper end - Add setting for proxy online mode status
diff --git a/src/main/java/net/minecraft/server/network/ServerCommonPacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerCommonPacketListenerImpl.java
index 02e65b0bd212d46855baee48fab35dc95a88b43f..bcd4f35a27c79fc3f02769ebedc1483d2cb8fa9e 100644
--- a/src/main/java/net/minecraft/server/network/ServerCommonPacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerCommonPacketListenerImpl.java
@@ -2,8 +2,10 @@ package net.minecraft.server.network;
 
 import com.mojang.authlib.GameProfile;
 import com.mojang.logging.LogUtils;
+
 import java.util.Objects;
 import javax.annotation.Nullable;
+
 import net.minecraft.ChatFormatting;
 import net.minecraft.CrashReport;
 import net.minecraft.CrashReportCategory;
@@ -32,7 +34,9 @@ import org.slf4j.Logger;
 
 // CraftBukkit start
 import io.netty.buffer.ByteBuf;
+
 import java.util.concurrent.ExecutionException;
+
 import org.bukkit.craftbukkit.entity.CraftPlayer;
 import org.bukkit.craftbukkit.util.CraftChatMessage;
 import org.bukkit.craftbukkit.util.CraftLocation;
@@ -68,6 +72,7 @@ public abstract class ServerCommonPacketListenerImpl implements ServerCommonPack
         this.player = player;
         this.cserver = minecraftserver.server;
     }
+
     protected final ServerPlayer player;
     protected final org.bukkit.craftbukkit.CraftServer cserver;
     public boolean processedDisconnect;
@@ -82,6 +87,7 @@ public abstract class ServerCommonPacketListenerImpl implements ServerCommonPack
         // Paper start - Fix kick event leave message not being sent
         this.onDisconnect(reason, null);
     }
+
     public void onDisconnect(Component reason, @Nullable net.kyori.adventure.text.Component quitMessage) {
         // Paper end - Fix kick event leave message not being sent
         if (this.isSingleplayerOwner()) {
@@ -120,7 +126,8 @@ public abstract class ServerCommonPacketListenerImpl implements ServerCommonPack
     }
 
     @Override
-    public void handlePong(ServerboundPongPacket packet) {}
+    public void handlePong(ServerboundPongPacket packet) {
+    }
 
     // CraftBukkit start
     private static final ResourceLocation CUSTOM_REGISTER = new ResourceLocation("register");
@@ -138,7 +145,7 @@ public abstract class ServerCommonPacketListenerImpl implements ServerCommonPack
         }
         PacketUtils.ensureRunningOnSameThread(packet, this, this.player.serverLevel());
         ResourceLocation identifier = packet.payload().id();
-        ByteBuf payload = ((ServerboundCustomPayloadPacket.UnknownPayload)packet.payload()).data();
+        ByteBuf payload = ((ServerboundCustomPayloadPacket.UnknownPayload) packet.payload()).data();
 
         if (identifier.equals(ServerCommonPacketListenerImpl.CUSTOM_REGISTER)) {
             try {
@@ -150,7 +157,7 @@ public abstract class ServerCommonPacketListenerImpl implements ServerCommonPack
                 ServerGamePacketListenerImpl.LOGGER.error("Couldn\'t register custom payload", ex);
                 this.disconnect("Invalid payload REGISTER!", org.bukkit.event.player.PlayerKickEvent.Cause.INVALID_PAYLOAD); // Paper - kick event cause
             }
-        // Purpur start
+            // Purpur start
         } else if (identifier.equals(PURPUR_CLIENT)) {
             try {
                 player.purpurClient = true;
@@ -241,21 +248,21 @@ public abstract class ServerCommonPacketListenerImpl implements ServerCommonPack
                 }
             }
         } else
-        // Purpur end
+            // Purpur end
 
-        if (this.keepAlivePending) {
-            if (!this.processedDisconnect && elapsedTime >= KEEPALIVE_LIMIT) { // check keepalive limit, don't fire if already disconnected
-                ServerGamePacketListenerImpl.LOGGER.warn("{} was kicked due to keepalive timeout!", this.player.getScoreboardName()); // more info
-                this.disconnect(ServerCommonPacketListenerImpl.TIMEOUT_DISCONNECTION_MESSAGE, org.bukkit.event.player.PlayerKickEvent.Cause.TIMEOUT); // Paper - kick event cause
-            }
-        } else {
-            if (elapsedTime >= 15000L) { // 15 seconds
-                this.keepAlivePending = true;
-                this.keepAliveTime = currentTime;
-                this.keepAliveChallenge = currentTime;
-                this.send(new ClientboundKeepAlivePacket(this.keepAliveChallenge));
+            if (this.keepAlivePending) {
+                if (!this.processedDisconnect && elapsedTime >= KEEPALIVE_LIMIT) { // check keepalive limit, don't fire if already disconnected
+                    ServerGamePacketListenerImpl.LOGGER.warn("{} was kicked due to keepalive timeout!", this.player.getScoreboardName()); // more info
+                    this.disconnect(ServerCommonPacketListenerImpl.TIMEOUT_DISCONNECTION_MESSAGE, org.bukkit.event.player.PlayerKickEvent.Cause.TIMEOUT); // Paper - kick event cause
+                }
+            } else {
+                if (elapsedTime >= 15000L) { // 15 seconds
+                    this.keepAlivePending = true;
+                    this.keepAliveTime = currentTime;
+                    this.keepAliveChallenge = currentTime;
+                    this.send(new ClientboundKeepAlivePacket(this.keepAliveChallenge));
+                }
             }
-        }
         // Paper end - give clients a longer time to respond to pings as per pre 1.12.2 timings
 
         //this.server.getProfiler().pop(); // Purpur
@@ -271,11 +278,27 @@ public abstract class ServerCommonPacketListenerImpl implements ServerCommonPack
     }
 
     public void send(Packet<?> packet) {
+        // Zirkonium start - No Chat Reports
+        if (this instanceof ServerGamePacketListenerImpl listener
+                && me.elephant1214.zirkonium.configuration.ZirkoniumConfig.convertToSystemMessage
+                && packet instanceof net.minecraft.network.protocol.game.ClientboundPlayerChatPacket chat) {
+            packet = new net.minecraft.network.protocol.game.ClientboundSystemChatPacket(io.papermc.paper.adventure.PaperAdventure.asAdventure(chat.chatType().resolve(listener.player.level()
+                    .registryAccess()).get().decorate(chat.unsignedContent() != null ? chat.unsignedContent()
+                    : Component.literal(chat.body().content()))), false);
+            listener.send(packet);
+            return;
+        }
         this.send(packet, (PacketSendListener) null);
     }
 
     public void send(Packet<?> packet, @Nullable PacketSendListener callbacks) {
-        // CraftBukkit start
+        if (this instanceof ServerGamePacketListenerImpl listener
+                && me.elephant1214.zirkonium.configuration.ZirkoniumConfig.convertToSystemMessage
+                && packet instanceof net.minecraft.network.protocol.game.ClientboundPlayerChatPacket chat && callbacks != null) {
+            listener.send(chat);
+            return;
+        }
+        // CraftBukkit start // Zirkonium end - No Chat Reports
         if (packet == null || this.processedDisconnect) { // Spigot
             return;
         } else if (packet instanceof ClientboundSetDefaultSpawnPositionPacket) {
@@ -299,7 +322,8 @@ public abstract class ServerCommonPacketListenerImpl implements ServerCommonPack
     }
 
     // CraftBukkit start
-    @Deprecated @io.papermc.paper.annotation.DoNotUse // Paper
+    @Deprecated
+    @io.papermc.paper.annotation.DoNotUse // Paper
     public void disconnect(String s) { // Paper
         this.disconnect(net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().deserialize(s), org.bukkit.event.player.PlayerKickEvent.Cause.UNKNOWN); // Paper
     }
@@ -311,7 +335,8 @@ public abstract class ServerCommonPacketListenerImpl implements ServerCommonPack
     }
 
     // Paper start
-    @Deprecated @io.papermc.paper.annotation.DoNotUse // Paper
+    @Deprecated
+    @io.papermc.paper.annotation.DoNotUse // Paper
     public void disconnect(final Component reason) {
         this.disconnect(io.papermc.paper.adventure.PaperAdventure.asAdventure(reason), org.bukkit.event.player.PlayerKickEvent.Cause.UNKNOWN);
     }
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 323391509c2efd49c54d7b36d7aa880c63ae5fd4..a8ca319d7411d64d9758807d526254d6808d9707 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -1,6 +1,5 @@
 package net.minecraft.server.players;
 
-import co.aikar.timings.MinecraftTimings;
 import com.google.common.collect.Lists;
 import com.google.common.collect.Maps;
 import com.google.common.collect.Sets;
@@ -109,7 +108,6 @@ import org.bukkit.Location;
 import org.bukkit.craftbukkit.CraftServer;
 import org.bukkit.craftbukkit.CraftWorld;
 import org.bukkit.craftbukkit.entity.CraftPlayer;
-import org.bukkit.craftbukkit.util.CraftChatMessage;
 import org.bukkit.craftbukkit.util.CraftLocation;
 import org.bukkit.entity.Player;
 import org.bukkit.event.player.PlayerChangedWorldEvent;
@@ -118,7 +116,6 @@ import org.bukkit.event.player.PlayerLoginEvent;
 import org.bukkit.event.player.PlayerQuitEvent;
 import org.bukkit.event.player.PlayerRespawnEvent;
 import org.bukkit.event.player.PlayerRespawnEvent.RespawnReason;
-import org.bukkit.event.player.PlayerSpawnChangeEvent;
 // CraftBukkit end
 
 public abstract class PlayerList {
@@ -1497,7 +1494,7 @@ public abstract class PlayerList {
     }
 
     public boolean verifyChatTrusted(PlayerChatMessage message) { // Paper - private -> public
-        return message.hasSignature() && !message.hasExpiredServer(Instant.now());
+        return me.elephant1214.zirkonium.configuration.ZirkoniumConfig.noChatReports || (message.hasSignature() && !message.hasExpiredServer(Instant.now())); // Zirkonium - No Chat Reports
     }
 
     // CraftBukkit start
