From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <46881115+Elephant1214@users.noreply.github.com>
Date: Tue, 6 Jun 2023 13:08:16 -0600
Subject: [PATCH] Implement configurable maximum anvil cost


diff --git a/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java b/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
index 62d7ba4e9d999b75c15ce72fc8683ed4dd2cec2b..5e54a277214bbd947b6a9efeb77cf08d3c615664 100644
--- a/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
+++ b/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
@@ -75,4 +75,10 @@ public class ZirkoniumConfig {
         addQueryData = getBoolean("no-chat-reports.add-query-data", addQueryData) && noChatReports;
         convertToSystemMessage = getBoolean("no-chat-reports.convert-to-system-message", convertToSystemMessage) && noChatReports;
     }
+    
+    public static int maxAnvilCost = 40;
+
+    private static void gameplaySettings() {
+        maxAnvilCost = getInt("gameplay.anvil.max-cost", maxAnvilCost);
+    }
 }
diff --git a/src/main/java/net/minecraft/world/inventory/AnvilMenu.java b/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
index 60260bcb568613b4a076385e9329a06f3e0fb885..567923b707e3641f83cea3b638bd129bd117237b 100644
--- a/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
+++ b/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
@@ -56,7 +56,7 @@ public class AnvilMenu extends ItemCombinerMenu {
     private static final int SLOT_Y_PLACEMENT = 47;
     // CraftBukkit start
     public static final int DEFAULT_DENIED_COST = -1;
-    public int maximumRepairCost = 40;
+    public int maximumRepairCost = me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxAnvilCost;
     private CraftInventoryView bukkitEntity;
     // CraftBukkit end
     // Purpur start - Anvil API
