From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <nathangetsmail@gmail.com>
Date: Sat, 24 Dec 2022 17:16:04 -0700
Subject: [PATCH] Implement a configurable maximum enchantment level multiplier


diff --git a/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java b/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
index bd53fba9b41e8fbfeb83fc73d28ee36a4598cb21..0df909dafbc72637f5164331967b12fe2631fea4 100644
--- a/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
+++ b/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
@@ -40,6 +40,7 @@ public class ZirkoniumConfig {
     }
 
     static void readConfig() {
+        Gameplay.gameplaySettings();
         NoChatReports.noChatReportsSettings();
 
         try {
@@ -68,6 +69,19 @@ public class ZirkoniumConfig {
         return config.getInt(path, config.getInt(path));
     }
 
+    public static class Gameplay {
+        public static int maxEnchantLevelMultiplier = 1;
+
+        public static void gameplaySettings() {
+            maxEnchantLevelMultiplier = getInt("gameplay.enchantment.maxLevelMultiplier", maxEnchantLevelMultiplier);
+            if (maxEnchantLevelMultiplier < 1) {
+                log(Level.WARNING, "The maximum enchantment level multiplier can not be less than 1, resetting it.");
+                maxEnchantLevelMultiplier = 1;
+                set("gameplay.enchantment.maxLevelMultiplier", 1);
+            }
+        }
+    }
+
     public static class NoChatReports {
         public static boolean noChatReports = true;
         public static boolean addQueryData = true;
diff --git a/src/main/java/net/minecraft/world/item/enchantment/ArrowDamageEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/ArrowDamageEnchantment.java
index 974300069c1e5b755e40ec3435da25ed62f484f4..bccfc459e732d494628ee3a94f501d90d8120b4f 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/ArrowDamageEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/ArrowDamageEnchantment.java
@@ -19,6 +19,6 @@ public class ArrowDamageEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 5;
+        return 5 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/ArrowKnockbackEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/ArrowKnockbackEnchantment.java
index b3feb7e9403adebbe42640a64b1b4b2dca9d6a10..34d453ed7bb1d8228fc008d9a864cb160a1a37f9 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/ArrowKnockbackEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/ArrowKnockbackEnchantment.java
@@ -19,6 +19,6 @@ public class ArrowKnockbackEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 2;
+        return 2 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/ArrowPiercingEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/ArrowPiercingEnchantment.java
index 853e9291f0869d33e4a35d5c291972d597c1314c..08c5e5fe287ecfd87649aab50ebe54c1e9e4de9d 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/ArrowPiercingEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/ArrowPiercingEnchantment.java
@@ -19,7 +19,7 @@ public class ArrowPiercingEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 4;
+        return 4 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/DamageEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/DamageEnchantment.java
index d6c354856c71d9629828e2ddbe316306bfaf6ea6..a8e8f60d5d8a6766419a2bbae7a8667b36813bf7 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/DamageEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/DamageEnchantment.java
@@ -37,7 +37,7 @@ public class DamageEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 5;
+        return 5 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/DigDurabilityEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/DigDurabilityEnchantment.java
index cbd39863b75245f1a5a16ab49f09238c034e09d2..ca0935e32eb9a9f3a7ce748042c5168d424531c0 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/DigDurabilityEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/DigDurabilityEnchantment.java
@@ -22,7 +22,7 @@ public class DigDurabilityEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 3;
+        return 3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/DiggingEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/DiggingEnchantment.java
index c55b492e9bcf5a9eef1bf86d9a0ada7231c459e9..3641c69058fd671bdf89b3aabbfebb209bf7c953 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/DiggingEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/DiggingEnchantment.java
@@ -21,7 +21,7 @@ public class DiggingEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 5;
+        return 5 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/FireAspectEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/FireAspectEnchantment.java
index 2b9ba92cf944e315b681f113d909fe5b2eb7cf1d..8cb47977dd34915018476aace2d0927834cab6a3 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/FireAspectEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/FireAspectEnchantment.java
@@ -19,6 +19,6 @@ public class FireAspectEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 2;
+        return 2 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/FishingSpeedEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/FishingSpeedEnchantment.java
index 83969917890d80c15b7c2ef87a3237b809edd2ae..c31d253a9ae03745512ef76c47ecfc56e199469a 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/FishingSpeedEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/FishingSpeedEnchantment.java
@@ -19,6 +19,6 @@ public class FishingSpeedEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 3;
+        return 3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/FrostWalkerEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/FrostWalkerEnchantment.java
index ba96da9869ac1b31a9311f2b4a92356f3dd3ab99..36ba527854b494227d08c39346999c56b6eb3dd9 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/FrostWalkerEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/FrostWalkerEnchantment.java
@@ -34,7 +34,7 @@ public class FrostWalkerEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 2;
+        return 2 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 
     public static void onEntityMoved(LivingEntity entity, Level world, BlockPos blockPos, int level) {
diff --git a/src/main/java/net/minecraft/world/item/enchantment/KnockbackEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/KnockbackEnchantment.java
index 52e02c8460ae53b9cb3d7f626d6204e1da9861a5..3d6a2f6e4f4a0dbe6959f223a5895f4c8f960289 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/KnockbackEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/KnockbackEnchantment.java
@@ -19,6 +19,6 @@ public class KnockbackEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 2;
+        return 2 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/LootBonusEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/LootBonusEnchantment.java
index 8fe09c13643d99639fb242da4367c42ef31b38b4..c85084122265bb11895220367de4a07e6d09d265 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/LootBonusEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/LootBonusEnchantment.java
@@ -27,7 +27,7 @@ public class LootBonusEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 3;
+        return 3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/OxygenEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/OxygenEnchantment.java
index 41e3f792b38cfd30dd5e8cd62e141e54b7d12219..e9801fc4458efe450d16bd23fbad4b3e31d6e884 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/OxygenEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/OxygenEnchantment.java
@@ -19,6 +19,6 @@ public class OxygenEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 3;
+        return 3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/ProtectionEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/ProtectionEnchantment.java
index ba8bd0df934f28c5dae6fd8a85ec0b73f6d3aa31..233e26f5cfe4e2d06be9c0ce37012a9be8fb076b 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/ProtectionEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/ProtectionEnchantment.java
@@ -26,7 +26,7 @@ public class ProtectionEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 4;
+        return 4 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/QuickChargeEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/QuickChargeEnchantment.java
index 44cf140303241ab2c646cd2b2b0a671abbb27325..b42d1c4f13e201fe540a827252d68c30b96a4054 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/QuickChargeEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/QuickChargeEnchantment.java
@@ -19,6 +19,6 @@ public class QuickChargeEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 3;
+        return 3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/SoulSpeedEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/SoulSpeedEnchantment.java
index 3d7797e8d11fab0c891cd78bb5d9d042a3165992..4b4d4aa63574ff3c54c654ba6df250a436a94da3 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/SoulSpeedEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/SoulSpeedEnchantment.java
@@ -34,6 +34,6 @@ public class SoulSpeedEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 3;
+        return 3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/SweepingEdgeEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/SweepingEdgeEnchantment.java
index ba80b900d58716bd7181cec4644966c56542b2d5..6655c7a30b4baf5d5d53872f5d4c0f29c7fad706 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/SweepingEdgeEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/SweepingEdgeEnchantment.java
@@ -19,7 +19,7 @@ public class SweepingEdgeEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 3;
+        return 3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 
     public static float getSweepingDamageRatio(int level) {
diff --git a/src/main/java/net/minecraft/world/item/enchantment/SwiftSneakEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/SwiftSneakEnchantment.java
index f9599f5b6d3050c106cb962a5924b1b58811f34f..b5ba872bb6caef27d2fbe2435e0d4e810ffcf316 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/SwiftSneakEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/SwiftSneakEnchantment.java
@@ -34,6 +34,6 @@ public class SwiftSneakEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 3;
+        return 3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/ThornsEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/ThornsEnchantment.java
index 618d645e71762ec8279640a60ea00460809bba63..aad181801d4883256c6bb5a8a6f68391b4e719ac 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/ThornsEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/ThornsEnchantment.java
@@ -27,7 +27,7 @@ public class ThornsEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 3;
+        return 3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/TridentImpalerEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/TridentImpalerEnchantment.java
index 62fc006eaa0cbcf96adfda5d211f1c04df34f4de..a48bfc9dc9abc500255c3e7d59fe44d2fee7c2b8 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/TridentImpalerEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/TridentImpalerEnchantment.java
@@ -20,7 +20,7 @@ public class TridentImpalerEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 5;
+        return 5 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/TridentLoyaltyEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/TridentLoyaltyEnchantment.java
index bacf68564b46e11d015e5e74d502bdcfa5b6fb55..6035ef83403ed402e4e6beab6815d79ba880770b 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/TridentLoyaltyEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/TridentLoyaltyEnchantment.java
@@ -19,6 +19,6 @@ public class TridentLoyaltyEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 3;
+        return 3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/TridentRiptideEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/TridentRiptideEnchantment.java
index 14315582296fa273248a7e5d56d6b9bb79645f1e..1764d307df6e6925f1d83318ef37fe8a940a1532 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/TridentRiptideEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/TridentRiptideEnchantment.java
@@ -19,7 +19,7 @@ public class TridentRiptideEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 3;
+        return 3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/WaterWalkerEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/WaterWalkerEnchantment.java
index 4ca8411508e0b2d9edf861ed92adbf07a5c6b3e7..daeea5fc99fa5a73a5de1f8959e857b3bdd7456e 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/WaterWalkerEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/WaterWalkerEnchantment.java
@@ -19,7 +19,7 @@ public class WaterWalkerEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return 3;
+        return 3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxEnchantLevelMultiplier;
     }
 
     @Override
