From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <46881115+Elephant1214@users.noreply.github.com>
Date: Sat, 24 Dec 2022 17:16:04 -0700
Subject: [PATCH] Implement a configurable maximum enchantment level multiplier


diff --git a/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java b/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
index 76ca77707a811c79a6b3a13d31ba1e9f49d20c08..fc11e40392705071a27cec9df50155e7af27cf92 100644
--- a/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
+++ b/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
@@ -42,6 +42,7 @@ public class ZirkoniumConfig {
 
     static void readConfig() {
         noChatReportsSettings();
+        gameplaySettings();
 
         try {
             config.save(CONFIG_FILE);
@@ -76,4 +77,15 @@ public class ZirkoniumConfig {
         addQueryData = getBoolean("no-chat-reports.add-query-data", addQueryData) && noChatReports;
         convertToSystemMessage = getBoolean("no-chat-reports.convert-to-system-message", convertToSystemMessage) && noChatReports;
     }
+
+    public static int maxEnchantLevelMultiplier = 1;
+
+    private static void gameplaySettings() {
+        maxEnchantLevelMultiplier = getInt("gameplay.enchantment.max-level-multiplier", maxEnchantLevelMultiplier);
+        if (maxEnchantLevelMultiplier < 1) {
+            LOGGER.warning("The maximum enchantment level multiplier cannot be less than 1, resetting it");
+            maxEnchantLevelMultiplier = 1;
+            set("gameplay.enchantment.max-level-multiplier", 1);
+        }
+    }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/Enchantments.java b/src/main/java/net/minecraft/world/item/enchantment/Enchantments.java
index 06ff34a333532c569d80919b702728d89e649969..7cfdb05381d8d48091d5418117fb8265e0436d70 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/Enchantments.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/Enchantments.java
@@ -132,7 +132,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.ARMOR_ENCHANTABLE),
                         10,
-                        4,
+                        4 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(1, 11),
                         Enchantment.dynamicCost(12, 11),
                         1,
@@ -155,7 +155,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.ARMOR_ENCHANTABLE),
                         5,
-                        4,
+                        4 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(10, 8),
                         Enchantment.dynamicCost(18, 8),
                         2,
@@ -191,7 +191,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.FOOT_ARMOR_ENCHANTABLE),
                         5,
-                        4,
+                        4 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(5, 6),
                         Enchantment.dynamicCost(11, 6),
                         2,
@@ -215,7 +215,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.ARMOR_ENCHANTABLE),
                         2,
-                        4,
+                        4 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(5, 8),
                         Enchantment.dynamicCost(13, 8),
                         4,
@@ -249,7 +249,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.ARMOR_ENCHANTABLE),
                         5,
-                        4,
+                        4 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(3, 6),
                         Enchantment.dynamicCost(9, 6),
                         2,
@@ -274,7 +274,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.HEAD_ARMOR_ENCHANTABLE),
                         2,
-                        3,
+                        3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(10, 10),
                         Enchantment.dynamicCost(40, 10),
                         4,
@@ -323,7 +323,7 @@ public class Enchantments {
                         holderGetter3.getOrThrow(ItemTags.ARMOR_ENCHANTABLE),
                         holderGetter3.getOrThrow(ItemTags.CHEST_ARMOR_ENCHANTABLE),
                         1,
-                        3,
+                        3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(10, 20),
                         Enchantment.dynamicCost(60, 20),
                         8,
@@ -348,7 +348,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.FOOT_ARMOR_ENCHANTABLE),
                         2,
-                        3,
+                        3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(10, 10),
                         Enchantment.dynamicCost(25, 10),
                         4,
@@ -373,7 +373,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.FOOT_ARMOR_ENCHANTABLE),
                         2,
-                        2,
+                        2 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(10, 10),
                         Enchantment.dynamicCost(25, 10),
                         4,
@@ -443,7 +443,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.FOOT_ARMOR_ENCHANTABLE),
                         1,
-                        3,
+                        3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(10, 10),
                         Enchantment.dynamicCost(25, 10),
                         8,
@@ -564,7 +564,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.LEG_ARMOR_ENCHANTABLE),
                         1,
-                        3,
+                        3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(25, 25),
                         Enchantment.dynamicCost(75, 25),
                         8,
@@ -589,7 +589,7 @@ public class Enchantments {
                         holderGetter3.getOrThrow(ItemTags.SHARP_WEAPON_ENCHANTABLE),
                         holderGetter3.getOrThrow(ItemTags.SWORD_ENCHANTABLE),
                         10,
-                        5,
+                        5 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(1, 11),
                         Enchantment.dynamicCost(21, 11),
                         1,
@@ -607,7 +607,7 @@ public class Enchantments {
                         holderGetter3.getOrThrow(ItemTags.WEAPON_ENCHANTABLE),
                         holderGetter3.getOrThrow(ItemTags.SWORD_ENCHANTABLE),
                         5,
-                        5,
+                        5 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(5, 8),
                         Enchantment.dynamicCost(25, 8),
                         2,
@@ -631,7 +631,7 @@ public class Enchantments {
                         holderGetter3.getOrThrow(ItemTags.WEAPON_ENCHANTABLE),
                         holderGetter3.getOrThrow(ItemTags.SWORD_ENCHANTABLE),
                         5,
-                        5,
+                        5 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(5, 8),
                         Enchantment.dynamicCost(25, 8),
                         2,
@@ -672,7 +672,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.SWORD_ENCHANTABLE),
                         5,
-                        2,
+                        2 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(5, 20),
                         Enchantment.dynamicCost(55, 20),
                         2,
@@ -689,7 +689,7 @@ public class Enchantments {
                         holderGetter3.getOrThrow(ItemTags.FIRE_ASPECT_ENCHANTABLE),
                         holderGetter3.getOrThrow(ItemTags.SWORD_ENCHANTABLE),
                         2,
-                        2,
+                        2 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(10, 20),
                         Enchantment.dynamicCost(60, 20),
                         4,
@@ -711,7 +711,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.SWORD_ENCHANTABLE),
                         2,
-                        3,
+                        3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(15, 9),
                         Enchantment.dynamicCost(65, 9),
                         4,
@@ -735,7 +735,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.SWORD_ENCHANTABLE),
                         2,
-                        3,
+                        3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(5, 9),
                         Enchantment.dynamicCost(20, 9),
                         4,
@@ -759,7 +759,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.MINING_ENCHANTABLE),
                         10,
-                        5,
+                        5 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(1, 10),
                         Enchantment.dynamicCost(51, 10),
                         1,
@@ -800,7 +800,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.DURABILITY_ENCHANTABLE),
                         5,
-                        3,
+                        3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(5, 8),
                         Enchantment.dynamicCost(55, 8),
                         2,
@@ -825,7 +825,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.MINING_LOOT_ENCHANTABLE),
                         2,
-                        3,
+                        3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(15, 9),
                         Enchantment.dynamicCost(65, 9),
                         4,
@@ -841,7 +841,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.BOW_ENCHANTABLE),
                         10,
-                        5,
+                        5 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(1, 10),
                         Enchantment.dynamicCost(16, 10),
                         1,
@@ -863,7 +863,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.BOW_ENCHANTABLE),
                         2,
-                        2,
+                        2 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(12, 20),
                         Enchantment.dynamicCost(37, 20),
                         4,
@@ -922,7 +922,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.FISHING_ENCHANTABLE),
                         2,
-                        3,
+                        3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(15, 9),
                         Enchantment.dynamicCost(65, 9),
                         4,
@@ -938,7 +938,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.FISHING_ENCHANTABLE),
                         2,
-                        3,
+                        3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(15, 9),
                         Enchantment.dynamicCost(65, 9),
                         4,
@@ -954,7 +954,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.TRIDENT_ENCHANTABLE),
                         5,
-                        3,
+                        3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(12, 7),
                         Enchantment.constantCost(50),
                         2,
@@ -970,7 +970,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.TRIDENT_ENCHANTABLE),
                         2,
-                        5,
+                        5 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(1, 8),
                         Enchantment.dynamicCost(21, 8),
                         4,
@@ -994,7 +994,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.TRIDENT_ENCHANTABLE),
                         2,
-                        3,
+                        3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(17, 7),
                         Enchantment.constantCost(50),
                         4,
@@ -1079,7 +1079,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.CROSSBOW_ENCHANTABLE),
                         5,
-                        3,
+                        3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(12, 20),
                         Enchantment.constantCost(50),
                         2,
@@ -1110,7 +1110,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.CROSSBOW_ENCHANTABLE),
                         10,
-                        4,
+                        4 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(1, 10),
                         Enchantment.constantCost(50),
                         1,
@@ -1127,7 +1127,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.MACE_ENCHANTABLE),
                         5,
-                        5,
+                        5 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(5, 8),
                         Enchantment.dynamicCost(25, 8),
                         2,
@@ -1144,7 +1144,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.MACE_ENCHANTABLE),
                         2,
-                        4,
+                        4 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(15, 9),
                         Enchantment.dynamicCost(65, 9),
                         4,
@@ -1161,7 +1161,7 @@ public class Enchantments {
                     Enchantment.definition(
                         holderGetter3.getOrThrow(ItemTags.MACE_ENCHANTABLE),
                         2,
-                        3,
+                        3 * me.elephant1214.zirkonium.configuration.ZirkoniumConfig.maxEnchantLevelMultiplier,
                         Enchantment.dynamicCost(15, 9),
                         Enchantment.dynamicCost(65, 9),
                         4,
