From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <nathangetsmail@gmail.com>
Date: Tue, 6 Jun 2023 13:08:16 -0600
Subject: [PATCH] Implement configurable maximum anvil cost


diff --git a/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java b/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
index 0df909dafbc72637f5164331967b12fe2631fea4..028cd33f0060dc7ac0927c4ffac387760b326ec9 100644
--- a/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
+++ b/src/main/java/me/elephant1214/zirkonium/configuration/ZirkoniumConfig.java
@@ -70,9 +70,11 @@ public class ZirkoniumConfig {
     }
 
     public static class Gameplay {
+        public static int maxAnvilCost = 40;
         public static int maxEnchantLevelMultiplier = 1;
 
         public static void gameplaySettings() {
+            maxAnvilCost = getInt("gameplay.anvil.maxCost", maxAnvilCost);
             maxEnchantLevelMultiplier = getInt("gameplay.enchantment.maxLevelMultiplier", maxEnchantLevelMultiplier);
             if (maxEnchantLevelMultiplier < 1) {
                 log(Level.WARNING, "The maximum enchantment level multiplier can not be less than 1, resetting it.");
diff --git a/src/main/java/net/minecraft/world/inventory/AnvilMenu.java b/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
index cb4a1cd39f12d90d30dd95efa2de58f57d9c6ee0..a2e875fe713d70aaf00e8045072f0a8fe46bd499 100644
--- a/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
+++ b/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
@@ -55,7 +55,7 @@ public class AnvilMenu extends ItemCombinerMenu {
     private static final int SLOT_Y_PLACEMENT = 47;
     // CraftBukkit start
     public static final int DEFAULT_DENIED_COST = -1;
-    public int maximumRepairCost = 40;
+    public int maximumRepairCost = me.elephant1214.zirkonium.configuration.ZirkoniumConfig.Gameplay.maxAnvilCost;
     private CraftInventoryView bukkitEntity;
     // CraftBukkit end
     public boolean bypassCost = false; // Purpur
