From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <nathangetsmail@gmail.com>
Date: Tue, 6 Jun 2023 13:08:16 -0600
Subject: [PATCH] Feature-Add configurable maximum anvil level cost


diff --git a/src/main/java/net/minecraft/world/inventory/AnvilMenu.java b/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
index 7ade5cd93b3d7b7259bf3246a8b74b0b31407817..d24b12b5c74ab798a3804e2afec18b38be12d84c 100644
--- a/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
+++ b/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
@@ -52,7 +52,7 @@ public class AnvilMenu extends ItemCombinerMenu {
     private static final int SLOT_Y_PLACEMENT = 47;
     // CraftBukkit start
     public static final int DEFAULT_DENIED_COST = -1;
-    public int maximumRepairCost = 40;
+    public int maximumRepairCost = nathan.zirkonium.configuration.ZirkoniumConfig.maxAnvilLevelCost;
     private CraftInventoryView bukkitEntity;
     // CraftBukkit end
     public boolean bypassCost = false; // Purpur
