From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <nathangetsmail@gmail.com>
Date: Thu, 6 Oct 2022 10:41:34 -0600
Subject: [PATCH] IDEA forgot how to wrap brackets


diff --git a/src/main/java/net/minecraft/world/inventory/AnvilMenu.java b/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
index 87bb65aa7b40052ae0e942ccfc253dbb4691c1a0..3ef5d8da6ccf3a863f2601021cc2568baae42bb7 100644
--- a/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
+++ b/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
@@ -233,10 +233,9 @@ public class AnvilMenu extends ItemCombinerMenu {
                                 flag2 = true;
                             } else {
                                 flag1 = true;
-                                // Zirkonium
-                                if ((!org.purpurmc.purpur.PurpurConfig.allowUnsafeEnchants || !org.purpurmc.purpur.PurpurConfig.allowHigherEnchantsLevels|| ZirkoniumConfig.maxEnchantmentLevelMultiplier > 1)
-                                        && i2 > enchantment.getMaxLevel()) { // Purpur // Zirkonium
+                                if ((!org.purpurmc.purpur.PurpurConfig.allowUnsafeEnchants || !org.purpurmc.purpur.PurpurConfig.allowHigherEnchantsLevels|| ZirkoniumConfig.maxEnchantmentLevelMultiplier > 1) && i2 > enchantment.getMaxLevel()) { // Purpur // Zirkonium
                                     i2 = enchantment.getMaxLevel();
+                                }
 
                                 map.put(enchantment, i2);
                                 int j2 = 0;
