From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <nathangetsmail@gmail.com>
Date: Tue, 14 Feb 2023 17:12:25 -0700
Subject: [PATCH] Fix Not Secure shown before messages


diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index c27e455321951e76e4818fec0e64301f5620dbf6..16c5596b3702192c67a0d7aa79501e7d1c88a281 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -1464,7 +1464,7 @@ public abstract class PlayerList {
     }
 
     public boolean verifyChatTrusted(PlayerChatMessage message) { // Paper - private -> public
-        return message.hasSignature() && !message.hasExpiredServer(Instant.now());
+        return nathan.zirkonium.configuration.ZirkoniumConfig.disableChatSigning || message.hasSignature() && !message.hasExpiredServer(Instant.now());
     }
 
     // CraftBukkit start
