From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <nathangetsmail@gmail.com>
Date: Tue, 14 Feb 2023 17:12:25 -0700
Subject: [PATCH] Fix Not Secure shown before messages


diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 82093c413722e280b9fcb46fb0ba03ca629e54fa..dbcc2ef3d987767b2758a6152c54bc28fd622f30 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -1450,7 +1450,7 @@ public abstract class PlayerList {
     }
 
     public boolean verifyChatTrusted(PlayerChatMessage message) { // Paper - private -> public
-        return message.hasSignature() && !message.hasExpiredServer(Instant.now());
+        return nathan.zirkonium.configuration.ZirkoniumConfig.disableChatSigning || message.hasSignature() && !message.hasExpiredServer(Instant.now());
     }
 
     // CraftBukkit start
