From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <nathangetsmail@gmail.com>
Date: Thu, 6 Oct 2022 11:05:47 -0600
Subject: [PATCH] Disable chat signing and max ench multiplier features, build
 fixes


diff --git a/build.gradle.kts b/build.gradle.kts
index 2ee34f92e8a89b90448bbf710fdc0d5d6350e919..5dc0c359ab34a193eb80430ac9d1f90da9962cc5 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -9,7 +9,7 @@ plugins {
 }
 
 dependencies {
-    implementation(project(":purpur-api")) // Purpur
+    implementation(project(":Zirkonium-API")) // Purpur // Zirkonium
     // Pufferfish start
     implementation("io.papermc.paper:paper-mojangapi:1.19.2-R0.1-SNAPSHOT") {
         exclude("io.papermc.paper", "paper-api")
