From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <nathangetsmail@gmail.com>
Date: Mon, 20 Mar 2023 10:42:07 -0600
Subject: [PATCH] Remove DoNotUse annotation to fix compilation 'error'


diff --git a/src/main/java/net/minecraft/network/protocol/game/ClientboundSystemChatPacket.java b/src/main/java/net/minecraft/network/protocol/game/ClientboundSystemChatPacket.java
index 1387e3597c43fd652f2fc82ca6fc2e83039604e2..e57b70b0f54ec5609398c8eb2c555f4053df4e3d 100644
--- a/src/main/java/net/minecraft/network/protocol/game/ClientboundSystemChatPacket.java
+++ b/src/main/java/net/minecraft/network/protocol/game/ClientboundSystemChatPacket.java
@@ -8,7 +8,7 @@ import net.minecraft.network.protocol.Packet;
 // Spigot start
 public record ClientboundSystemChatPacket(@javax.annotation.Nullable net.kyori.adventure.text.Component adventure$content, @javax.annotation.Nullable String content, boolean overlay) implements Packet<ClientGamePacketListener> { // Paper - Adventure
 
-    @io.papermc.paper.annotation.DoNotUse // Paper - No locale context
+    // @io.papermc.paper.annotation.DoNotUse // Paper - No locale context // Zirkonium - I'll use whatever I want, thank you very much
     public ClientboundSystemChatPacket(Component content, boolean overlay) {
         this(null, Component.Serializer.toJson(content), overlay); // Paper - Adventure
     }
