From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <nathangetsmail@gmail.com>
Date: Thu, 6 Oct 2022 15:54:15 -0600
Subject: [PATCH] Fixed another mistake but with enchantments this time


diff --git a/src/main/java/net/minecraft/world/inventory/AnvilMenu.java b/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
index a67025c621526dbded5bb5c0702a07f19f9f44d4..5771a3348e271b45b5a71b23b2ed5e41d2ba7168 100644
--- a/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
+++ b/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
@@ -4,7 +4,6 @@ import com.mojang.logging.LogUtils;
 import java.util.Iterator;
 import java.util.Map;
 
-import nathan.zirkonium.configuration.ZirkoniumConfig;
 import net.minecraft.network.chat.Component;
 import net.minecraft.tags.BlockTags;
 import net.minecraft.world.entity.player.Inventory;
diff --git a/src/main/java/net/minecraft/world/item/enchantment/ArrowDamageEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/ArrowDamageEnchantment.java
index 7f8bf2d6e1f394fc9a7ecd0063416ee1150e94cf..79b3bf63a4cddc011531cb715fbb4152cc7c3423 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/ArrowDamageEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/ArrowDamageEnchantment.java
@@ -21,6 +21,6 @@ public class ArrowDamageEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 5 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 5;
+        return 5 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/ArrowKnockbackEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/ArrowKnockbackEnchantment.java
index 814eb28ae10f992101ffd88ba33824bf99db1c7a..25c157c86c1a5e45466f01fa82356d1744559829 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/ArrowKnockbackEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/ArrowKnockbackEnchantment.java
@@ -21,6 +21,6 @@ public class ArrowKnockbackEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 2 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 2;
+        return 2 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/ArrowPiercingEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/ArrowPiercingEnchantment.java
index edb6af052ff507db558fa44c382ed761d92608f6..cee5ea49448d549df77292e77ebc3c926002e972 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/ArrowPiercingEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/ArrowPiercingEnchantment.java
@@ -21,7 +21,7 @@ public class ArrowPiercingEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 4 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 4;
+        return 4 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/DamageEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/DamageEnchantment.java
index 07f155c27c59bae154fc4f6e43d3ddb4bef7a131..63186829c8dbde9b46d5536df988eb724abc2016 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/DamageEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/DamageEnchantment.java
@@ -39,7 +39,7 @@ public class DamageEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 5 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 5;
+        return 5 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/DigDurabilityEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/DigDurabilityEnchantment.java
index 296cd5f712892c5cef8244ffaa789f3e517ffd19..a9ea0238903e3b1d4c774b77d98425b878df846d 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/DigDurabilityEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/DigDurabilityEnchantment.java
@@ -24,7 +24,7 @@ public class DigDurabilityEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 3;
+        return 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/DiggingEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/DiggingEnchantment.java
index 6b21fa406242c7385c8ccaf22f4e2210334809b5..d8b7bbd3d2290c7fd240977599e2f7bfabe59cd8 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/DiggingEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/DiggingEnchantment.java
@@ -23,7 +23,7 @@ public class DiggingEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 5 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 5;
+        return 5 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/FireAspectEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/FireAspectEnchantment.java
index 6a6d67cf04d501ea0df171d6f8f49c2d34f9e165..d88c91e8e1e36e142f2ccd90162d77e2892d5208 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/FireAspectEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/FireAspectEnchantment.java
@@ -21,6 +21,6 @@ public class FireAspectEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 2 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 2;
+        return 2 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/FishingSpeedEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/FishingSpeedEnchantment.java
index 1564b33b6a2166a59a280e15c9a5311033ff439f..e96638a97b03f762ead1bc69fa198ce3117f25fc 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/FishingSpeedEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/FishingSpeedEnchantment.java
@@ -21,6 +21,6 @@ public class FishingSpeedEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 3;
+        return 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/FrostWalkerEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/FrostWalkerEnchantment.java
index ae43e9c0016d02875839eeaec5cf191f8be353cb..b4b3485de77f7cd600609e27de064d4202e390a4 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/FrostWalkerEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/FrostWalkerEnchantment.java
@@ -38,7 +38,7 @@ public class FrostWalkerEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 2 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 2;
+        return 2 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 
     public static void onEntityMoved(LivingEntity entity, Level world, BlockPos blockPos, int level) {
diff --git a/src/main/java/net/minecraft/world/item/enchantment/KnockbackEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/KnockbackEnchantment.java
index b8f27de228db9898dc451e9c7f5c204c154dfc45..56899fae968950739ae6f4017eb1fcdb4bbbf0df 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/KnockbackEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/KnockbackEnchantment.java
@@ -21,6 +21,6 @@ public class KnockbackEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 2 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 2;
+        return 2 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/LootBonusEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/LootBonusEnchantment.java
index da7abe4ee8b382dcefe4c5d224b036f41e0b5b0a..b9980b04620d086f22c90ec62fa8fcd1044734d7 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/LootBonusEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/LootBonusEnchantment.java
@@ -29,7 +29,7 @@ public class LootBonusEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 3;
+        return 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/OxygenEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/OxygenEnchantment.java
index d0410ff5778f03f40300469f7ca92f540d9ab6f5..fa6f0f1b6be518290ad5bd3d6119068693ec306b 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/OxygenEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/OxygenEnchantment.java
@@ -21,6 +21,6 @@ public class OxygenEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 3;
+        return 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/ProtectionEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/ProtectionEnchantment.java
index 2ad93cfd046b212856a80b89fb7074f3ea1665ec..f6099fde4b807959b6f4fb678fef6d630355c32d 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/ProtectionEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/ProtectionEnchantment.java
@@ -27,7 +27,7 @@ public class ProtectionEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 4 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 4;
+        return 4 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/QuickChargeEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/QuickChargeEnchantment.java
index 6a28b3acd3148743ebb98251ea55ce688f3bd90a..b6aa4a719bf00fbc681d563a2103de266aecca13 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/QuickChargeEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/QuickChargeEnchantment.java
@@ -21,6 +21,6 @@ public class QuickChargeEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 3;
+        return 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/SoulSpeedEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/SoulSpeedEnchantment.java
index b95b7011a4f7c535398cd3ca10ed45a70a35dae2..b1a3bd4d59a25d1c9eb549850732044498f5eacf 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/SoulSpeedEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/SoulSpeedEnchantment.java
@@ -36,6 +36,6 @@ public class SoulSpeedEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 3;
+        return 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/SweepingEdgeEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/SweepingEdgeEnchantment.java
index 315a1bcab99cfd0df37bce99c8a09bcb55625982..c9454ec7ff94549f5a975bcdc0080c88fbf4ea7d 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/SweepingEdgeEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/SweepingEdgeEnchantment.java
@@ -21,7 +21,7 @@ public class SweepingEdgeEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 3;
+        return 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 
     public static float getSweepingDamageRatio(int level) {
diff --git a/src/main/java/net/minecraft/world/item/enchantment/SwiftSneakEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/SwiftSneakEnchantment.java
index 5b5aeed6f46b6654da4c66007231d730593739fc..4928e489a337bcde08c564fd080be06772aa2d82 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/SwiftSneakEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/SwiftSneakEnchantment.java
@@ -36,6 +36,6 @@ public class SwiftSneakEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 3;
+        return 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/ThornsEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/ThornsEnchantment.java
index 472caaa3326b81d06d5569849b548674fda7c975..862355ddf635d1068c78631605a3016e29a6a000 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/ThornsEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/ThornsEnchantment.java
@@ -31,7 +31,7 @@ public class ThornsEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 3;
+        return 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/TridentImpalerEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/TridentImpalerEnchantment.java
index c0032f9a16da8a5da38e7e3ed9058d33eb1f2756..ca3db7975ae7863738b384cc7db6791dc0216967 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/TridentImpalerEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/TridentImpalerEnchantment.java
@@ -22,7 +22,7 @@ public class TridentImpalerEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 5 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 5;
+        return 5 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/TridentLoyaltyEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/TridentLoyaltyEnchantment.java
index 503ae624c5f1077d36fa9fd4d80efeef281d934e..fcea92208864119c068501bd8636879fd2233299 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/TridentLoyaltyEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/TridentLoyaltyEnchantment.java
@@ -21,6 +21,6 @@ public class TridentLoyaltyEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 3;
+        return 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 }
diff --git a/src/main/java/net/minecraft/world/item/enchantment/TridentRiptideEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/TridentRiptideEnchantment.java
index a058984f957651301ee0e842d98b0e94c14bff0c..17cd215100f330faa1553e67c74583c4dcc06608 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/TridentRiptideEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/TridentRiptideEnchantment.java
@@ -21,7 +21,7 @@ public class TridentRiptideEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 3;
+        return 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/item/enchantment/WaterWalkerEnchantment.java b/src/main/java/net/minecraft/world/item/enchantment/WaterWalkerEnchantment.java
index 4e96f52524ecf3ce98baef38abee46bf653e4de4..1f133bbba9dd317a445b02cd894ec2b71417fcdc 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/WaterWalkerEnchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/WaterWalkerEnchantment.java
@@ -21,7 +21,7 @@ public class WaterWalkerEnchantment extends Enchantment {
 
     @Override
     public int getMaxLevel() {
-        return PurpurConfig.allowHigherEnchantsLevels ? 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier : 3;
+        return 3 * ZirkoniumConfig.maxEnchantmentLevelMultiplier;
     }
 
     @Override
