From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <nathangetsmail@gmail.com>
Date: Sat, 8 Oct 2022 13:57:23 -0600
Subject: [PATCH] Feature-anvilMaxLevelMultiplier


diff --git a/src/main/java/nathan/zirkonium/configuration/ZirkoniumConfig.java b/src/main/java/nathan/zirkonium/configuration/ZirkoniumConfig.java
index c3b37b0a79c2914deaaa4e02ef702268ce56bc0d..00171cc78c7173599553cd313c7df0952ca9a604 100644
--- a/src/main/java/nathan/zirkonium/configuration/ZirkoniumConfig.java
+++ b/src/main/java/nathan/zirkonium/configuration/ZirkoniumConfig.java
@@ -72,9 +72,11 @@ public class ZirkoniumConfig {
         return config.getInt(path, config.getInt(path));
     }
 
+    public static int anvilMaxLevelMultiplier = 1;
     public static int maxEnchantmentLevelMultiplier = 1;
     @SuppressWarnings("unused")
     public static void enchantmentSettings() {
+        anvilMaxLevelMultiplier = getInt("settings.enchantments.anvil-max-level-multiplier", anvilMaxLevelMultiplier);
         maxEnchantmentLevelMultiplier = getInt("settings.enchantments.max-enchantment-level-multiplier", maxEnchantmentLevelMultiplier);
         if (maxEnchantmentLevelMultiplier < 1) {
             log(Level.WARNING, "Maximum enchantment level multiplier can not be less than one, resetting it");
diff --git a/src/main/java/net/minecraft/world/inventory/AnvilMenu.java b/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
index 5771a3348e271b45b5a71b23b2ed5e41d2ba7168..35316375aece009e7c883e85094784d1acd19173 100644
--- a/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
+++ b/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
@@ -46,7 +46,7 @@ public class AnvilMenu extends ItemCombinerMenu {
     private static final int COST_RENAME = 1;
     // CraftBukkit start
     public static final int DEFAULT_DENIED_COST = -1;
-    public int maximumRepairCost = 40;
+    public int maximumRepairCost = 40 * nathan.zirkonium.configuration.ZirkoniumConfig.anvilMaxLevelMultiplier;
     private CraftInventoryView bukkitEntity;
     // CraftBukkit end
     public boolean bypassCost = false; // Purpur
@@ -77,8 +77,8 @@ public class AnvilMenu extends ItemCombinerMenu {
         ItemStack itemstack = activeQuickItem == null ? stack : activeQuickItem; // Purpur
         if (org.purpurmc.purpur.event.inventory.AnvilTakeResultEvent.getHandlerList().getRegisteredListeners().length > 0) new org.purpurmc.purpur.event.inventory.AnvilTakeResultEvent(player.getBukkitEntity(), getBukkitView(), org.bukkit.craftbukkit.inventory.CraftItemStack.asCraftMirror(itemstack)).callEvent(); // Purpur
         if (!player.getAbilities().instabuild) {
-            if (bypassCost) ((ServerPlayer) player).lastSentExp = -1; else // Purpur
-            player.giveExperienceLevels(-this.cost.get());
+            if (bypassCost) ((ServerPlayer) player).lastSentExp = -1;
+            else player.giveExperienceLevels(-this.cost.get()); // Purpur
         }
 
         this.inputSlots.setItem(0, ItemStack.EMPTY);
