From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <nathangetsmail@gmail.com>
Date: Sat, 8 Oct 2022 14:50:12 -0600
Subject: [PATCH] Change anvil level config option


diff --git a/src/main/java/nathan/zirkonium/configuration/ZirkoniumConfig.java b/src/main/java/nathan/zirkonium/configuration/ZirkoniumConfig.java
index 00171cc78c7173599553cd313c7df0952ca9a604..fe257aee2c64ab9c6c7a7e5b1bc5c56b64fd6082 100644
--- a/src/main/java/nathan/zirkonium/configuration/ZirkoniumConfig.java
+++ b/src/main/java/nathan/zirkonium/configuration/ZirkoniumConfig.java
@@ -72,11 +72,11 @@ public class ZirkoniumConfig {
         return config.getInt(path, config.getInt(path));
     }
 
-    public static int anvilMaxLevelMultiplier = 1;
+    public static int maxAnvilLevelCost = 40;
     public static int maxEnchantmentLevelMultiplier = 1;
     @SuppressWarnings("unused")
     public static void enchantmentSettings() {
-        anvilMaxLevelMultiplier = getInt("settings.enchantments.anvil-max-level-multiplier", anvilMaxLevelMultiplier);
+        maxAnvilLevelCost = getInt("settings.enchantments.max-anvil-level-cost", maxAnvilLevelCost);
         maxEnchantmentLevelMultiplier = getInt("settings.enchantments.max-enchantment-level-multiplier", maxEnchantmentLevelMultiplier);
         if (maxEnchantmentLevelMultiplier < 1) {
             log(Level.WARNING, "Maximum enchantment level multiplier can not be less than one, resetting it");
diff --git a/src/main/java/net/minecraft/world/inventory/AnvilMenu.java b/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
index 35316375aece009e7c883e85094784d1acd19173..fef2d17d8252e37a817f5b2c0f8023551cef141a 100644
--- a/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
+++ b/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
@@ -46,7 +46,7 @@ public class AnvilMenu extends ItemCombinerMenu {
     private static final int COST_RENAME = 1;
     // CraftBukkit start
     public static final int DEFAULT_DENIED_COST = -1;
-    public int maximumRepairCost = 40 * nathan.zirkonium.configuration.ZirkoniumConfig.anvilMaxLevelMultiplier;
+    public int maximumRepairCost = nathan.zirkonium.configuration.ZirkoniumConfig.maxAnvilLevelCost;
     private CraftInventoryView bukkitEntity;
     // CraftBukkit end
     public boolean bypassCost = false; // Purpur
