From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nathan <nathangetsmail@gmail.com>
Date: Wed, 12 Oct 2022 22:06:55 -0400
Subject: [PATCH] Add fix for plugins changing anvil level to below set level


diff --git a/src/main/java/nathan/zirkonium/configuration/ZirkoniumConfig.java b/src/main/java/nathan/zirkonium/configuration/ZirkoniumConfig.java
index fe257aee2c64ab9c6c7a7e5b1bc5c56b64fd6082..2e47897ad593005fc60625c14081a27062dd9e73 100644
--- a/src/main/java/nathan/zirkonium/configuration/ZirkoniumConfig.java
+++ b/src/main/java/nathan/zirkonium/configuration/ZirkoniumConfig.java
@@ -7,9 +7,6 @@ import org.bukkit.configuration.file.YamlConfiguration;
 
 import java.io.File;
 import java.io.IOException;
-import java.lang.reflect.InvocationTargetException;
-import java.lang.reflect.Method;
-import java.lang.reflect.Modifier;
 import java.util.logging.Level;
 
 @SuppressWarnings({"SameParameterValue", "deprecation"})
@@ -73,10 +70,12 @@ public class ZirkoniumConfig {
     }
 
     public static int maxAnvilLevelCost = 40;
+    public static boolean maxCannotBeLower = false;
     public static int maxEnchantmentLevelMultiplier = 1;
     @SuppressWarnings("unused")
     public static void enchantmentSettings() {
         maxAnvilLevelCost = getInt("settings.enchantments.max-anvil-level-cost", maxAnvilLevelCost);
+        maxCannotBeLower = getBoolean("settings.enchantments.max-cannot-be-lower", maxCannotBeLower);
         maxEnchantmentLevelMultiplier = getInt("settings.enchantments.max-enchantment-level-multiplier", maxEnchantmentLevelMultiplier);
         if (maxEnchantmentLevelMultiplier < 1) {
             log(Level.WARNING, "Maximum enchantment level multiplier can not be less than one, resetting it");
diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftInventoryAnvil.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftInventoryAnvil.java
index 27dd4eb4781a3c75772860c11db886e1038cecd2..6da9e0cd9cfb1d3bf9a955bd4124ac03b178f4e8 100644
--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftInventoryAnvil.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftInventoryAnvil.java
@@ -1,6 +1,7 @@
 package org.bukkit.craftbukkit.inventory;
 
 import com.google.common.base.Preconditions;
+import nathan.zirkonium.configuration.ZirkoniumConfig;
 import net.minecraft.world.Container;
 import net.minecraft.world.inventory.AnvilMenu;
 import org.bukkit.Location;
@@ -52,11 +53,17 @@ public class CraftInventoryAnvil extends CraftResultInventory implements AnvilIn
         return container.maximumRepairCost;
     }
 
+    // Zirkonium start
     @Override
     public void setMaximumRepairCost(int levels) {
-        Preconditions.checkArgument(levels >= 0, "Maximum repair cost must be positive (or 0)");
+        if (ZirkoniumConfig.maxCannotBeLower) {
+            Preconditions.checkArgument(levels >= ZirkoniumConfig.maxAnvilLevelCost, "Maximum repair cost must be more than or equal to " + ZirkoniumConfig.maxAnvilLevelCost);
+        } else {
+            Preconditions.checkArgument(levels >= 0, "Maximum repair cost must be positive (or 0)");
+        }
         container.maximumRepairCost = levels;
     }
+    // Zirkonium end
 
     // Purpur start
     @Override
