From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Elephant1214 <46881115+Elephant1214@users.noreply.github.com>
Date: Tue, 8 Jul 2025 12:07:21 -0600
Subject: [PATCH] Implement Sneaky


diff --git a/src/main/java/com/destroystokyo/paper/network/StandardPaperServerListPingEventImpl.java b/src/main/java/com/destroystokyo/paper/network/StandardPaperServerListPingEventImpl.java
index 30a19d10869f73d67b794e8e4c035bc5c10209e6..9a1c3f1f619190514c907a1b71df68d8ef29d010 100644
--- a/src/main/java/com/destroystokyo/paper/network/StandardPaperServerListPingEventImpl.java
+++ b/src/main/java/com/destroystokyo/paper/network/StandardPaperServerListPingEventImpl.java
@@ -96,10 +96,32 @@ public final class StandardPaperServerListPingEventImpl extends PaperServerListP
         } else {
             favicon = Optional.empty();
         }
-        final ServerStatus ping = new ServerStatus(description, players, Optional.of(version), favicon, server.enforceSecureProfile());
+        ServerStatus ping = new ServerStatus(description, players, Optional.of(version), favicon, server.enforceSecureProfile()); // Zirkonium - Not final
+        // Zirkonium start - Sneaky - Hide query data
+        if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.sneaky && !com.adryd.sneaky.IPList.INSTANCE.canPing(event.getClient().getAddress())) {
+            if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.hideServerPingData) {
+                ping = FAKE_STATUS;
+            } else if (me.elephant1214.zirkonium.configuration.ZirkoniumConfig.onlyHidePlayerList) {
+                ping = new ServerStatus(
+                    ping.description(),
+                    Optional.of(new ServerStatus.Players(ping.players().get().max(), 0, java.util.List.of())),
+                    ping.version(),
+                    ping.favicon(),
+                    ping.enforcesSecureChat()
+                );
+            }
+        } // Zirkonium end - Sneaky - Hide query data
 
         // Send response
         networkManager.send(new ClientboundStatusResponsePacket(ping));
     }
 
+    // Zirkonium start - Sneaky - Hide query data
+    private static final ServerStatus FAKE_STATUS = new ServerStatus(
+        Component.literal("A Minecraft Server"),
+        Optional.of(new ServerStatus.Players(0, 20, java.util.List.of())),
+        Optional.of(ServerStatus.Version.current()),
+        Optional.empty(),
+        true
+    ); // Zirkonium end - Sneaky - Hide data spoofing
 }
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 42645ea0f97df8e733001e7c3f4c4096ed6a5049..bcc28ef678f8ce370a53280173efc1519ff0dc50 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1067,6 +1067,7 @@ public final class CraftServer implements Server {
         this.console.paperConfigurations.reloadConfigs(this.console);
         this.console.galeConfigurations.reloadConfigs(this.console); // Gale - Gale configuration
         org.purpurmc.purpur.PurpurConfig.init((File) console.options.valueOf("purpur-settings")); // Purpur - Purpur config files
+        me.elephant1214.zirkonium.configuration.ZirkoniumConfig.init((File) console.options.valueOf("zirkonium-settings")); // Zirkonium - Zirkonium config
         for (ServerLevel world : this.console.getAllLevels()) {
             // world.serverLevelData.setDifficulty(config.difficulty); // Paper - per level difficulty
             world.setSpawnSettings(world.serverLevelData.getDifficulty() != Difficulty.PEACEFUL && config.spawnMonsters); // Paper - per level difficulty (from MinecraftServer#setDifficulty(ServerLevel, Difficulty, boolean))
diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index c03726447fa79270d00f3c4384f0b43f9e421073..42ea74ef00c570156d099d9ecce9f4fdb72bd48f 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -172,6 +172,14 @@ public class Main {
                     .describedAs("Yml file");
                 // Purpur end - Purpur config files
 
+                // Zirkonium start - Zirkonium config
+                this.acceptsAll(asList("zirkonium", "zirkonium-settings"), "File for Zirkonium settings")
+                    .withRequiredArg()
+                    .ofType(File.class)
+                    .defaultsTo(new File("zirkonium.yml"))
+                    .describedAs("Yml file");
+                // Zirkonium end - Zirkonium config
+
                 this.accepts("server-name", "Name of the server")
                         .withRequiredArg()
                         .ofType(String.class)
